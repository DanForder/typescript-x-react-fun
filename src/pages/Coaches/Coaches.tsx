import { Fragment, useEffect, useState } from "react";
import Layout from "../../components/Layout/Layout";
import { getCoaches } from "../../data/api";
import CoachData from "../../types/server/CoachData";
import { formatName } from "../../utils/nameUtils";

// to see this page component action, run the app "/coaches" appended to the URL 🧐
const Coaches = () => {
  const [coaches, setCoaches] = useState<CoachData[] | null>();

  useEffect(() => {
    (async () => {
      setCoaches((await getCoaches()) ?? null);
    })();
  }, []);

  const coachesTsx = (coaches: CoachData[] | null) => {
    return coaches && coaches.length > 0 ? (
      coaches.map((coach) => (
        <Fragment key={coach.id}>
          <h2>{formatName(coach.firstName, coach.lastName)}</h2>
          <p>Coach Id: {coach.id}</p>
        </Fragment>
      ))
    ) : (
      <span>Sorry, no coaches found</span>
    );
  };

  return (
    <Layout navigation>
      <h1>Coaches</h1>
      {coaches !== undefined ? coachesTsx(coaches) : <span>Loading...</span>}
    </Layout>
  );
};

export default Coaches;
