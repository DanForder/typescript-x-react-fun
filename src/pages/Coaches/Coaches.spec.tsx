import { customRender } from "../../utils/testUtils";
import Coaches from "./Coaches";

export default describe("Coaches Page", () => {
  it("Renders component in loading state", () => {
    const { container } = customRender(<Coaches />);
    expect(container).toMatchSnapshot();
  });
});
