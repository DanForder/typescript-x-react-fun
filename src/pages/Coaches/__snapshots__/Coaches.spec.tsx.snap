// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coaches Page Renders component in loading state 1`] = `
<div>
  <div
    class="layout"
  >
    <main
      class="layout__main"
    >
      <h1>
        Coaches
      </h1>
      <span>
        Loading...
      </span>
    </main>
  </div>
</div>
`;
